### ISSUE TITLE
Launching 25-NOV Build Causes Frequent & Prolonged Freezing Of Win10 OS+Frequent Disconnects

### ISSUE THREAD

**Issue by DeathStalker77** [nicotine-plus/nicotine-plus#2770]
Nicotine+ version:  
Win 10 Ent
Newst Build 25-NOV

### Describe the bug
Causes FREQUENT & PROLONGED **SYSTEM** freezing!  Not the same as the pausing/freezing of the app previously reported - this is a complete freeze/pause of the whole Windows system!  This is _especially_ pronounced on launching the app.  Also noticed frequent disconnects.

### Expected behavior
The app should not adversely affect normal operating system functioning

### Steps to reproduce the bug
Launch program under Win10 Enterprise, latest build of both app & OS.

### Additional context
N/A - not something that can be captured in a screenshot

**Comment by DeathStalker77** [comment]
Dropping back to the 06-NOV build, I noticed that it also causes the same type of freezing/pausing of the OS, but _not_ to the extent that the 25-NOV build does.

**Comment by mathiascode** [comment]
Have not noticed this on my end. Anything going on in the Windows task manager when the freezes happen?

**Comment by DeathStalker77** [comment]
I can't tell, because Windows is completely non-responsive.

**Comment by DeathStalker77** [comment]
I dropped back to the 06-NOV build I was previously using, and while this IS still present as the program launches, it is not as severe as the current version.  I don't know if this might have something to do with the size of one share, the number of shares, or the number of users/files queued for download - or a combination of any of them.

**Comment by mathiascode** [comment]
Could you try the builds below? Scroll down on the pages for download links. Please try all the builds until the freezes start happening, so we can pinpoint the issue.

November 10: https://github.com/nicotine-plus/nicotine-plus/actions/runs/6828995603
November 13: https://github.com/nicotine-plus/nicotine-plus/actions/runs/6846354280
November 17: https://github.com/nicotine-plus/nicotine-plus/actions/runs/6903803045
November 18: https://github.com/nicotine-plus/nicotine-plus/actions/runs/6915101473
November 24: https://github.com/nicotine-plus/nicotine-plus/actions/runs/6984482548
November 25: https://github.com/nicotine-plus/nicotine-plus/actions/runs/6986389035

**Comment by DeathStalker77** [comment]
@mathiascode - just a suggestion, but maybe add the BUILD# &/or Date to the ABOUT info as well? :-)

**Comment by DeathStalker77** [comment]
Just installed the 30-NOV build and while there IS still some freezing during launch, it is not nearly as bad as previously.  Will update on the status of the disconnects and any other issues after a while :-)

**Comment by mathiascode** [comment]
> Just installed the 30-NOV build and while there IS still some freezing during launch, it is not nearly as bad as previously. Will update on the status of the disconnects and any other issues after a while :-)

That's good to hear. Would you say it's back on the same level as the November 6 build?

Now if I could figure out #2745...

> just a suggestion, but maybe add the BUILD# &/or Date to the ABOUT info as well? :-)

It would be nice to have this eventually, but it isn't trivial to add it to all the development builds on different platforms.

**Comment by mathiascode** [comment]
> Yes, it is about the same as the 06-NOV build

Nice :)

 I do not know if it is worse with a different tab, as I usually start up on the default DOWNLOADS tab to begin with.

Could you check when you have the opportunity? Would be nice to know if the performance of the backend is fine at this point. The treeviews in the GUI have performance issues when appending a lot of rows at once, not much that can be done until they are ported to the newer Gtk.ColumnView widget.

**Comment by DeathStalker77** [comment]
I can tell you that I'm just under 120k files queued for DL (I'm also going to add Feature Requests for Total # of Files & Total MB/GB/TB queued).  Current max files queued from a single user is just over 50k.  Performance, once fully launched, seems to be pretty good :-)  I did note that there is a _bit_ of a delay when adding files from Browsing a User's list and switching over to the DOWNLOAD tab - it takes a second or 2 for the selected files to appear in the list.  That said, I wouldn't consider this to be a significant issue in the least, and it would seem that it might even improve, if not be eliminated, when the GTK issue has been resolved.
